castle.go

```diff
@@ -9,29 +9,38 @@ import (
 	"github.com/tomasen/realip"
 )
 
-// TrackEndpoint defines the tracking URL castle.io side
-var TrackEndpoint = "https://api.castle.io/v1/track"
+// FilterEndpoint defines the filter URL castle.io side
+var FilterEndpoint = "https://api.castle.io/v1/filter"
 
-// AuthenticateEndpoint defines the adaptive authentication URL castle.io side
-var AuthenticateEndpoint = "https://api.castle.io/v1/authenticate"
+// RiskEndpoint defines the risk URL castle.io side
+var RiskEndpoint = "https://api.castle.io/v1/risk"
 
-// Event is an enum defining types of event castle tracks
-type Event string
+type Event struct {
+	EventType   EventType
+	EventStatus EventStatus
+}
+
+// EventType is an enum defining types of event castle tracks
+type EventType string
+
+// See https://docs.castle.io/docs/events
+const (
+	EventTypeLogin                EventType = "$login"
+	EventTypeRegistration         EventType = "$registration"
+	EventTypeProfileUpdate        EventType = "$profile_update"
+	EventTypeProfileReset         EventType = "$profile_reset"
+	EventTypePasswordResetRequest EventType = "$password_reset_request"
+	EventTypeChallenge            EventType = "$challenge"
+)
+
+// EventStatus is an enum defining the statuses for a given event.
+type EventStatus string
 
-// See https://castle.io/docs/events
+// See https://docs.castle.io/docs/events
 const (
-	EventLoginSucceeded                Event = "$login.succeeded"
-	EventLoginFailed                   Event = "$login.failed"
-	EventPasswordResetRequestSucceeded Event = "$password_reset_request.succeeded"
-	EventPasswordResetRequestFailed    Event = "$password_reset_request.failed"
-	EventPasswordResetSucceeded        Event = "$password_reset.succeeded"
-	EventPasswordResetFailed           Event = "$password_reset.failed"
-	EventIncidentMitigated             Event = "$incident.mitigated"
-	EventReviewResolved                Event = "$review.resolved"
-	EventReviewEscalated               Event = "$review.escalated"
-	EventChallengeRequested            Event = "$challenge.requested"
-	EventChallengeSucceeded            Event = "$challenge.succeeded"
-	EventChallengeFailed               Event = "$challenge.failed"
+	EventStatusAttempted EventStatus = "$attempted"
```

Comment:
Perhaps we should add `$requested` as well to enable people to use the endpoints with all the type?